lib/components/user/trip-basics-pane.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react'
+import { ButtonToolbar, ControlLabel, FormControl, FormGroup, ToggleButton, ToggleButtonGroup } from 'react-bootstrap'
+
+import { arrayToDayFields, dayFieldsToArray } from '../../util/monitored-trip'
+import TripSummary from './trip-summary'
+
+class TripBasicsPane extends Component {
+  _handleTripDaysChange = e => {
+    const { onMonitoredTripChange } = this.props
+    onMonitoredTripChange(arrayToDayFields(e))
+  }
+
+  _handleTripNameChange = e => {
+    const { onMonitoredTripChange } = this.props
+    onMonitoredTripChange({ tripName: e.target.value })
+  }
+
+  render () {
+    const { monitoredTrip } = this.props
+    const { itinerary } = monitoredTrip
+
+    if (!itinerary) {
+      return <div>No itinerary to display.</div>
+    } else {
+      return (
+        <div>
+          <ControlLabel>Selected itinerary:</ControlLabel>
+          <TripSummary monitoredTrip={monitoredTrip} />
+
+          <FormGroup>
+            <ControlLabel>Please provide a name for this trip:</ControlLabel>
+            <FormControl
+              onChange={this._handleTripNameChange}
+              type='text'
+              value={monitoredTrip.tripName}
+            />
+          </FormGroup>
+
+          <FormGroup>
+            <ControlLabel>What days to you take this trip?</ControlLabel>
+            <ButtonToolbar>
+              <ToggleButtonGroup
+                onChange={this._handleTripDaysChange}
+                type='checkbox'
+                value={dayFieldsToArray(monitoredTrip)}
```

Comment:
Does this need to be updated for recent changes to otp middleware?

Reply:
The functions `dayFieldsToArray` and `arrayToDayFields` take care of the conversion from fields to array for the `ToggleButtonGroup`.